### PR TITLE
Move available port resolving to transport and cosmetic code changes

### PR DESCRIPTION
**Pull request by antonkharenko**

### CODE REVIEW

File: cluster/src/main/java/io/scalecube/cluster/Cluster.java [hunk 1]
```diff
@@ -141,19 +100,37 @@ public static ICluster joinAwait(ClusterConfig config) {
...
+        transport = input;
+
+        // Init gossip protocol component
+        gossipProtocol = new GossipProtocol(memberId, transport, config.gossipProtocolConfig);
+
+        // Init failure detector component
+        failureDetector = new FailureDetector(transport, config.failureDetectorConfig);
+
+        // Init cluster membership component
+        clusterMembership = new ClusterMembership(memberId, transport);
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/Cluster.java]: Why ``ClusterMembership`` does not accept ``config`` like  FD and Gossip components?

File: cluster/src/main/java/io/scalecube/cluster/Cluster.java [hunk 2]
```diff
@@ -141,19 +100,37 @@ public static ICluster joinAwait(ClusterConfig config) {
...
+        clusterMembership.setFailureDetector(failureDetector);
+        clusterMembership.setGossipProtocol(gossipProtocol);
+        clusterMembership.setLocalMetadata(config.metadata);
+        clusterMembership.setSeedMembers(config.seedMembers);
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/Cluster.java]: Next setters better to be removed :   ``setSyncTime``, ``setSyncTimeout``, ``setSyncGroup``, ``setMaxSuspectTime``, ``setMaxShutdownTime`` . Use ``config`` instead .

File: cluster/src/main/java/io/scalecube/cluster/fdetector/FailureDetector.java [hunk 3]
```diff
@@ -309,37 +290,28 @@ private void declareSuspected(Address member) {
...
 
-  private Address selectPingMember(List<Address> members) {
-    if (pingMember != null) {
-      return pingMember;
-    }
-    return members.isEmpty() ? null : selectRandomMembers(members, 1, null).get(0);
+  private Address selectPingMember() {
+    // TODO [AK]: Select ping member not randomly, but sequentially with shuffle after last member selected
+    return !members.isEmpty() ? members.get(ThreadLocalRandom.current().nextInt(members.size())) : null;
   }
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/fdetector/FailureDetector.java]: What was wrong with ``selectRandomMembers()`` name and code itself?

File: cluster/src/main/java/io/scalecube/cluster/gossip/IGossipProtocol.java [hunk 4]
```diff
@@ -12,10 +15,19 @@
  */
 public interface IGossipProtocol {
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/gossip/IGossipProtocol.java]: Following ``start()`` , ``stop()``  methods doesn't need live here, they can be moved to implementation class. Same applies to ``IFailureDetector`` start/stop.

File: cluster/src/main/java/io/scalecube/cluster/Cluster.java (hunk 1 above)
Comment (antonkharenko) [inline cluster/src/main/java/io/scalecube/cluster/Cluster.java]: It should, just not part of this PR.

File: cluster/src/main/java/io/scalecube/cluster/Cluster.java (hunk 2 above)
Comment (antonkharenko) [inline cluster/src/main/java/io/scalecube/cluster/Cluster.java]: Also part of Membership refactoring I plan to do.

File: cluster/src/main/java/io/scalecube/cluster/fdetector/FailureDetector.java (hunk 3 above)
Comment (antonkharenko) [inline cluster/src/main/java/io/scalecube/cluster/fdetector/FailureDetector.java]: select ping member should be actually not random, as I wrote in TODO - in order to have stable upper bound on failure detection time.

And then it is not just selectRandomMembers, but has specific meaning as part of the algorithm (selecting ping request members).

File: cluster/src/main/java/io/scalecube/cluster/gossip/IGossipProtocol.java (hunk 4 above)
Comment (antonkharenko) [inline cluster/src/main/java/io/scalecube/cluster/gossip/IGossipProtocol.java]: It depends, if I generally want to have possibility to replace failure detection algorithm with something else - then I need start/stop methods on interface, if not then I don't need interface here at all... not decided here yet.
